### PR TITLE
Fix identity assignment when labels are changed 

### PR DESCRIPTION
**Pull request by aanm**
**Summary of changes**:

- Re assign identity based on OpLabels
- Create new identity when endpoint labels change

```release-note
create new identity when endpoint labels change and re assign identity based on all endpoint labels when restoring
```

Reported by @mrdima

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]
LGTM!

File: pkg/endpoint/endpoint.go [hunk 1]
```diff
@@ -1768,7 +1768,9 @@ func (e *Endpoint) identityLabelsChanged(owner Owner, myChangeRev int) error {
 		return nil
 	}
 
-	if e.SecurityIdentity != nil && string(e.SecurityIdentity.Labels.SortedList()) != string(newLabels.SortedList()) {
+	if e.SecurityIdentity != nil &&
```
Comment (tgraf) [inline pkg/endpoint/endpoint.go:1771]: Am I missing the change here or is this just a cosmetic change?

File: daemon/state.go [hunk 2]
```diff
@@ -246,25 +246,21 @@ func readEPsFromDirNames(basePath string, eptsDirNames []string) []*endpoint.End
...
-	if ep.SecurityIdentity == nil {
-		return fmt.Errorf("Endpoint doesn't have a security label.")
-	}
-
 	// Filter the restored labels with the new daemon's filter
-	l, _ := labels.FilterLabels(ep.SecurityIdentity.Labels)
-	ep.SecurityIdentity.Labels = l
-
-	identity, _, err := identityPkg.AllocateIdentity(ep.SecurityIdentity.Labels)
+	l, _ := labels.FilterLabels(ep.OpLabels.AllLabels())
```
Comment (tgraf) [inline daemon/state.go]: This also includes disabled labels. That doesn't seem right. I think you want to use `e.OpLabels.IdentityLabels`

**Review by tgraf — CHANGES_REQUESTED** [review]

File: pkg/endpoint/endpoint.go (hunk 1 above)
Comment (aanm) [reply pkg/endpoint/endpoint.go:1771]: `!=` -> `==`

File: daemon/state.go (hunk 2 above)
Comment (aanm) [reply daemon/state.go]: oh, awesome, that's exactly what I want.

**Comment by aanm** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**